pkg/bmh/list.go

```diff
@@ -75,6 +75,55 @@ func List(apiClient *clients.Settings, nsname string, options ...goclient.ListOp
 	return bmhObjects, nil
 }
 
+// ListInAllNamespaces lists the BareMetalHosts across all namespaces on the provided cluster.
+func ListInAllNamespaces(apiClient *clients.Settings, options ...goclient.ListOptions) ([]*BmhBuilder, error) {
```

Comment:
In order to reduce code duplication please add private function "list" with common code which will be used as helper function in List and ListInAllNamespaces funcs.